modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/sourcer/object/ObjectSourcerImpl.java

```diff
@@ -0,0 +1,157 @@
+import javax.servlet.http.HttpServletRequest;
+
+import javax.ws.rs.core.UriInfo;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Luis Miguel Barcos
+ */
+@Component(service = Sourcer.class)
+public class ObjectSourcerImpl implements Sourcer {
+
+	@Override
+	public Map<String, PropertyInfo> getPropertiesInfo(
+		String entityName, Map<String, Schema> propertySchemas) {
+
+		Map<String, PropertyInfo> stringObjectPropertyMap = new HashMap<>();
+
+		for (Map.Entry<String, Schema> schemaEntry :
+				propertySchemas.entrySet()) {
+
+			Schema value = schemaEntry.getValue();
+
+			FieldDefinition fieldDefinition = value.getFieldDefinition();
+
+			String objectFieldName = fieldDefinition.getName();
+
+			stringObjectPropertyMap.put(
+				schemaEntry.getKey(),
+				new PropertyInfo(
+					entityName, objectFieldName, schemaEntry.getKey()));
+		}
+
+		return stringObjectPropertyMap;
+	}
+
+	@Override
+	public Object getValue(
+			PropertyInfo propertyInfo, Object pathParameterValue,
+			HttpServletRequest httpServletRequest, UriInfo uriInfo)
+		throws Exception {
+
+		long objectEntryId = 0;
+
+		if (pathParameterValue instanceof Long) {
+			objectEntryId = (long)pathParameterValue;
+		}
+
+		String internalClass = propertyInfo.getInternalClass();
+
+		long objectDefinitionId = GetterUtil.getLong(
+			internalClass.split("#")[1]);
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				objectDefinitionId);
+
+		DefaultObjectEntryManager defaultObjectEntryManager =
+			DefaultObjectEntryManagerProvider.provide(
+				_objectEntryManagerRegistry.getObjectEntryManager(
+					objectDefinition.getStorageType()));
+
+		return _getFieldValue(
+			defaultObjectEntryManager.getObjectEntry(
+				_getDTOConverterContext(
+					objectEntryId, objectDefinition, uriInfo,
+					httpServletRequest),
+				objectDefinition, objectEntryId),
+			propertyInfo.getInternalName());
+	}
+
+	private Object _getCustomFieldValue(
+		Map<String, Object> values, String name) {
+
+		return values.get(name);
+	}
+
+	private DTOConverterContext _getDTOConverterContext(
+			long objectEntryId, ObjectDefinition objectDefinition,
+			UriInfo uriInfo, HttpServletRequest httpServletRequest)
+		throws Exception {
+
+		return new DefaultDTOConverterContext(
+			false, null, _dtoConverterRegistry, httpServletRequest,
+			objectEntryId,
+			LocaleUtil.fromLanguageId(objectDefinition.getDefaultLanguageId()),
+			uriInfo, _userLocalService.getUser(objectDefinition.getUserId()));
+	}
+
+	private Object _getFieldValue(ObjectEntry objectEntry, String name) {
+		Object systemField = _getSystemFieldValue(objectEntry, name);
+
+		if (systemField == null) {
+			return _getCustomFieldValue(objectEntry.getProperties(), name);
+		}
+
+		return systemField;
+	}
+
+	private Object _getSystemFieldValue(ObjectEntry objectEntry, String name) {
+		if (name.equals("createDate")) {
```

Comment:
It is a bit confusing to provide the `createDate` field, as in the Objects API we provide the `dateCreated` field but in the Objects service the field is `createDate`. If we consider `createDate` for this purpose, would it be easier to check the `system` flag in the `ObjectField` table given a name?